crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -262,8 +262,14 @@ where
 
 #[async_trait]
 pub(crate) trait DigestHandler<A> {
+    type DigestResult: Future<Output = SuiResult>;
+
```

Comment:
The idea would be to change this to:

```suggestion
    type Error: std::error::Error;
    type Future: Future<Output = Result<(), Self::Error>>;
```

And then the caller can do the map themselves.

Reply:
I tried this and it got pretty awkward, so I went with using BoxFuture instead, which removes a lot of boilerplate.